gm4_noisy_blocks/data/noisy_blocks/functions/pulse_check.mcfunction

```diff
@@ -0,0 +1,29 @@
+#This function runs every tick. It plays a sound for anyone with an interaction score of 1 and immediately resets their score.
+execute at @a[scores={gm4_nbk_crafting=1..}] run playsound minecraft:entity.shulker.close block @a[distance=..8] ~ ~ ~ 0.5 1.4
+scoreboard players reset @a[scores={gm4_nbk_crafting=1..}] gm4_nbk_crafting
```

Comment:
Do you think for all of these it would be more performant to have a function per sound and reset the player's score in the function so we're halving the player tests?